freqtrade/freqtradebot.py

```diff
@@ -692,23 +701,25 @@ def _notify_enter(self, trade: Trade, order: Dict, order_type: Optional[str] = N
             'trade_id': trade.id,
             'type': RPCMessageType.BUY_FILL if fill else RPCMessageType.BUY,
             'buy_tag': trade.buy_tag,
-            'exchange': self.exchange.name.capitalize(),
+            'exchange': trade.exchange.capitalize(),
             'pair': trade.pair,
             'limit': open_rate,  # Deprecated (?)
             'open_rate': open_rate,
             'order_type': order_type,
             'stake_amount': trade.stake_amount,
             'stake_currency': self.config['stake_currency'],
             'fiat_currency': self.config.get('fiat_display_currency', None),
-            'amount': safe_value_fallback(order, 'filled', 'amount') or trade.amount,
+            'amount': order.get('filled') if fill else order.get('amount'),
```

Comment:
Why would you change this? 
the prior line was the "safe" version - which made sure a value was sent.

This will fail if filled = None in the order dict - as you'd pass None to the object.

Reply:
Updated